python/lsst/resources/s3.py

```diff
@@ -307,6 +317,35 @@ def _download_file(
             translate_client_error(err, self)
             raise
 
+    def to_fsspec(self) -> tuple[AbstractFileSystem, str]:
+        """Return an abstract file system and path that can be used by fsspec.
+
+        Returns
+        -------
+        fs : `fsspec.spec.AbstractFileSystem`
+            A file system object suitable for use with the returned path.
+        path : `str`
+            A path that can be opened by the file system object.
+        """
+        if s3fs is None:
+            raise ImportError("s3fs is not available")
+        # Must remove the profile from the URL and form it again.
+        endpoint_config = _get_s3_connection_parameters(self._profile)
+        s3 = s3fs.S3FileSystem(
+            profile=endpoint_config.profile,
+            endpoint_url=endpoint_config.endpoint_url,
+            key=endpoint_config.access_key_id,
+            secret=endpoint_config.secret_access_key,
+        )
+        if not _s3_validate_bucket():
+            # Force the boto client to be constructed so we can modify it.
+            with contextlib.suppress(FileNotFoundError):
+                s3.open("bad_bucket/path")
+            # This uses a private property for now.
+            s3._s3.meta.events.unregister("before-parameter-build.s3", validate_bucket_name)
```

Comment:
`S3FileSystem` has a public `s3` property you can use to create and get the boto client.

```suggestion
            s3.s3.meta.events.unregister("before-parameter-build.s3", validate_bucket_name)
```

Reply:
Thanks. I feel a bit better about a public property.